libs/api/src/lib/graphql/mutation.public.ts

```diff
@@ -554,6 +557,11 @@ export const GraphQLPublicMutation = new GraphQLObjectType<undefined, Context>({
           throw new SubscriptionNotFound()
         }
 
+        // Prevent user from extending not extendable subscription
+        if (!subscription.extendable) {
```

Comment:
A test for this extend-function as well as for the update function would be perfect. Could catch missing checks from other programmers, not involved with this functionality.